src/components/TreeGrid/TreeGrid.tsx

```diff
@@ -0,0 +1,254 @@
+            }
+        }
+        if (columnIndex === 1) {
+            return this.renderHiddenTreeCell(key, columnIndex, rowID);     
+        }
+        if (columnIndex === 2 && shouldIndent) {
+                style = {...style, width: style.width - indent};
+        }      
+        if (shouldIndent) {
+            style = {...style, left: style.left + indent};
+        }
+        if (columnIndex === 0) {
+            return this.renderActionTreeCell(key, rowIndex, rowData, style);
+        }
+        return this.renderBodyCell(columnIndex, key, rowIndex, rowData, style);     
+    }
+
+    renderHiddenTreeCell(key, columnIndex, rowData) {
+        const style = {
+            display: 'none'
+        };
+        return (
+        <div key={key} style={style}> 
+            {rowData}
+        </div>);
+    }
+
+    renderActionTreeCell(key, rowIndex: number, rowData: TreeNode, style) {
+        let actionsTooltip = rowData.IsExpanded ? 'Collapse' : 'Expand';
+        let iconName = rowData.IsExpanded ? 'icon-arrow_down' : 'icon-arrow_right';
+        let icon = null;
+        
+        if (rowData.leaves.length <= 0 && rowData.IsExpanded) {
+            icon = null;
+            actionsTooltip = null;
+        } else {
+            icon = <Icon iconName={iconName} className="expand-collapse-action-icon" />;
+        }
+        const rowClass = 'grid-row-' + rowIndex;
+        const onMouseEnter = () => { this.onMouseEnterCell(rowClass); };
+        const onMouseLeave = () => { this.onMouseLeaveCell(rowClass); };
+        const onToggleTreeRow = () => { this.onTreeExpandToggleClick(rowData); };
+        const title = actionsTooltip;
+        const className = classNames(
+            'grid-component-cell',
+            rowClass,
+            { 'is-selected': rowIndex === this.state.selectedRowIndex }
+        );
+        return (
+            <div
+                key={key}
+                style={style}
+                className={className}
+                onMouseEnter={onMouseEnter}
+                onMouseLeave={onMouseLeave}
+                title={title}
+                onClick={icon ? onToggleTreeRow : null}
+            >
+                {icon}
+            </div>
+        );
+    }
+
+    renderBodyCell(columnIndex: number, key, rowIndex: number, rowData, style) {
+        const columns = this.state.columnsToDisplay;
+        const notLastIndex = columnIndex < (columns.length - 1);
+        const column = columns[columnIndex];
+        const dataKey = column.dataMember || column.valueMember;
+        const cellData = rowData[dataKey];
+        const rowClass = 'grid-row-' + rowIndex;
+        const className = classNames(
+            'grid-component-cell',
+            rowClass,
+            column.cellClassName,
+            { 'border-column-cell': notLastIndex },
+            { 'is-selected': rowIndex === this.state.selectedRowIndex });
+
+        const onMouseEnter = () => { this.onMouseEnterCell(rowClass); };
```

Comment:
These arrow function may cause unnecessary rerenders.
You could avoid them by using special component for the div you are returning and binding these functions.